code/ship/shiphit.cpp

```diff
@@ -2409,6 +2410,8 @@ static void ship_do_damage(object *ship_objp, object *other_obj, vec3d *hitpos,
 		// apply damage to subsystems, and get back any remaining damage that needs to go to the hull
 		damage = do_subobj_hit_stuff(ship_objp, other_obj, hitpos, submodel_num, damage, &apply_hull_armor);
 
+		ets_update_max_speed(ship_objp);
+
```

Comment:
I would be as specific as possible and check directly for an engine subsystem being damaged to under the threshold inside the function. The case where we *know for certain* the max speed *will* change, not just whenever the ship takes damage.

The idea is to allow scripts to mess with velocity as much as possible, and only override it when we absolutely must.